autoload/battery/backend/freebsd.vim

```diff
@@ -0,0 +1,38 @@
+" Reference:
+" sysctl -d hw.acpi.battery
+"
+" hw.acpi.battery: battery status and info
+" hw.acpi.battery.info_expire: time in seconds until info is refreshed
+" hw.acpi.battery.units: number of batteries
+" hw.acpi.battery.state: current status flags
+"   discharging = 1
+"   charging = 2
+"   critical = 4
+" hw.acpi.battery.rate: present rate in mW
+" hw.acpi.battery.time: remaining time in minutes
+" hw.acpi.battery.life: percent capacity remaining
+
+let s:bat_status = 'hw.acpi.battery.state'
+let s:bat_capacity = 'hw.acpi.battery.life'
+
+function! s:read_sysctl(ctl) abort
+  let cmd = 'sysctl -n ' . a:ctl
+  return system(cmd)
```

Comment:
Please use `System.Job` instead to make it asynchronous. See other implementations or https://github.com/lambdalisue/vital-Whisky/blob/master/doc/Vital/System/Job.txt

Reply:
Maybe https://github.com/lambdalisue/battery.vim/pull/13 would help

Reply:
Done, though I had to realize `sysctl` output is hard to parse